Loop/Managers/NotificationManager.swift

```diff
@@ -80,34 +80,20 @@ extension NotificationManager {
 
     // MARK: - Notifications
 
-    static func sendBolusFailureNotification(for error: Error, units: Double, at startDate: Date) {
+    static func sendBolusFailureNotification(for error: PumpManagerError, units: Double, at startDate: Date) {
```

Comment:
Is this something that should be using the new `Alert`system?

Reply:
Yes, I thought a lot about this. I think it'd be possible, but it requires some special handling that alerts don't currently support.

1) This alert needs to stop whatever else is being presented; one of the design aspects was that the BG should be visible in the background while the alert is present.  This is a halt all other interaction type of alert.
2) If the user navigates dismisses the recovery view (i.e. taps "back"), then the alert should still be presented on the main screen, and block any other interaction.

If you think the alert system would be up for dealing with these kinds of requirements, we could talk about what changes we'd need to the alert protocols to support that.

Reply:
> This alert needs to stop whatever else is being presented; one of the design aspects was that the BG should be visible in the background while the alert is present. This is a halt all other interaction type of alert.

Hm...if that's the case, then how is this current use of a `UNUserNotification` going to stop all interaction?  I'm confused.

> If the user navigates dismisses the recovery view (i.e. taps "back"), then the alert should still be presented on the main screen, and block any other interaction.

Okay... but how does the user tap "back" if all other interaction is halted?  Again, sorry if I'm misunderstanding...